File: Algorithms/Implementations/manasa_and_stones.c
#include <stdio.h>

int combinations_possible(int number_of_stones, int a, int b)
{
    // calculate the possible number of combination
    int max, min, diff;
    if (a < b)
    {
        max = b;
        min = a;
        diff = b - a;
        return ((max - min) / diff * number_of_stones);
    } else if (a > b)
    {
        max = a;
        min = b;
        diff = a -b;
        return ((max - min) / diff * number_of_stones);
    } else {
        return 1;
    }
        
}

void possible_numbers(int number_of_stones, int a, int b)
{
    // calculate the possible number of combination
    int combinations_count = combinations_possible(number_of_stones, a, b);
    int combination[combinations_count];

    int all_as = (number_of_stones - 1) * a;
    int all_bs = (number_of_stones - 1) * b;
    int min_number, max_number, difference;
    if (all_as < all_bs)
    {
        min_number = all_as;
        max_number = all_bs;
        difference = b - a;
    } else if (all_as > all_bs)
    {
        min_number = all_bs;
        max_number = all_as;
        difference = a - b;
    } else {
        min_number = all_as;
        max_number = all_bs;
        difference = a - b;
    }
    for (int j = 0; j < combinations_count; j++)
    {
        combination[j] = min_number;
        min_number+= difference;
    }
    for (int i=0; i < combinations_count; i++)
    {
        printf("%d ", combination[i]);
    }
    printf("\n");
}

int main()
{
    int test_count;

    scanf("%d", &test_count);

    int number_of_stones[test_count];
    int a[test_count];
    int b[test_count];

    for (int i = 0; i < test_count; i++)
    {
        scanf("%d", &number_of_stones[i]);
        scanf("%d", &a[i]);
        scanf("%d", &b[i]);
    }

    for (int j = 0; j < test_count; j++)
    {
        possible_numbers(number_of_stones[j], a[j], b[j]);
    }
    return 0;
}

File: Algorithms/Implementations/manasa_and_stones.py

def possible_numbers(number_of_stones, a, b):
    all_as = (number_of_stones - 1) * a
    all_bs = (number_of_stones -1) * b
    min_number = min(all_as, all_bs)
    max_number = max(all_as, all_bs)
    number = min_number
    all_numbers = [min_number, max_number]
    difference = abs(a - b)
    if difference > 0:
        while number < max_number - difference:
            number += difference
            all_numbers.append(number)
        all_numbers.sort()
        return all_numbers
    else:
        return [max_number]
        

def brute_force_possible_numbers(number_of_stones, a, b, iteration=0, stones=[0]):
    if iteration == number_of_stones - 1:
        stones.sort()
        return stones
    else:
        new_set_of_stones = []
        for stone in stones:
            new_set_of_stones.append(stone + a)
            new_set_of_stones.append(stone + b)
    return possible_numbers(number_of_stones, a, b, iteration + 1, list(set(new_set_of_stones)))

if __name__ == "__main__":
    test_count = int(raw_input().strip())
    for i in xrange(test_count):
        number_of_stones = int(raw_input().strip())
        a = int(raw_input().strip())
        b = int(raw_input().strip())

        print " ".join(map(str, possible_numbers(number_of_stones, a, b)))
